recipe_scrapers/_utils.py

```diff
@@ -202,17 +221,15 @@ def url_path_to_dict(path):
         r"^"
         r"((?P<schema>.+?)://)?"
         r"((?P<user>.+?)(:(?P<password>.*?))?@)?"
-        r"(?P<host>.*?)"
-        r"(:(?P<port>\d+?))?"
+        r"(?P<host>[^:/]+)"
+        r"(:(?P<port>\d+))?"
```

Comment:
Nitpick: it may seem strange, but I think an empty port number is valid.  It could be worth checking the relevant RFC ([RFC3986](https://www.rfc-editor.org/rfc/rfc3986) if I remember correctly) to double-check that.

```suggestion
        r"(:(?P<port>\d+?))?"
```